src/Geta.NotFoundHandler.Admin/Areas/GetaNotFoundHandlerAdmin/Pages/Administer.cshtml.cs

```diff
@@ -166,21 +205,43 @@ public IActionResult OnPostImportDeletedRedirects()
         });
     }
 
-    public IActionResult OnPostExportRedirects()
+    public IActionResult OnPostExportRedirects(string TypeId)
     {
-        var redirects = _redirectsService.GetSaved().ToList();
-        var document = _redirectsXmlParser.Export(redirects);
+        if (!string.IsNullOrEmpty(TypeId) && TypeId == "xml")
```

Comment:
1) Check if typeId is null or empty and return BuildResponse(string message, CardType cardType).
2) Check if the redirects collection has any items; if not, return BuildResponse(string message, CardType cardType).
3) Create two methods, ExportAsXml and ExportAsCsv, and use them in a switch statement.